### PR TITLE
fix(tasks) Change update_user_reports to use bulk queries

### PR DESCRIPTION
**Pull request by evanh**
I noticed that sometimes this task was failing with timeouts/rate limit errors,
so change it to use bulk queries to hopefully avoid this problem.

### CODE REVIEW

File: src/sentry/tasks/update_user_reports.py [hunk 1]
```diff
@@ -19,16 +19,29 @@ def update_user_reports(**kwargs):
...
-    updated = 0
-    for report in user_reports:
-        count += 1
-        event = eventstore.get_event_by_id(report.project_id, report.event_id)
-        if event:
+    project_ids = [r.project_id for r in user_reports]
+    event_ids = [r.event_id for r in user_reports]
+    report_by_event = {r.event_id: r for r in user_reports}
+
+    snuba_filter = eventstore.Filter(project_ids=project_ids, event_ids=event_ids)
```
Comment (markstory) [inline src/sentry/tasks/update_user_reports.py]: You might want to do 1 query per project. The primary key for events only ensures that event_id is unique *within* a project. While it *should* be globally unique there is a chance it won't be and you're update will do the wrong thing.

Comment (evanh) [reply src/sentry/tasks/update_user_reports.py]: Gotcha. OK will do.

File: src/sentry/tasks/update_user_reports.py [hunk 2]
```diff
@@ -19,16 +20,43 @@ def update_user_reports(**kwargs):
...
-    count = 0
-    updated = 0
-    for report in user_reports:
-        count += 1
-        event = eventstore.get_event_by_id(report.project_id, report.event_id)
-        if event:
-            report.update(group_id=event.group_id, environment=event.get_environment())
-            updated += 1
+    # We do one query per project, just to avoid the small case that two projects have the same event ID
+    project_map = {}
```
Comment (markstory) [inline src/sentry/tasks/update_user_reports.py]: You could use `collections.defaultdict` to save a setdefault() call, but it is more of a🏌 level comment.

File: src/sentry/tasks/update_user_reports.py [hunk 3]
```diff
@@ -19,16 +20,43 @@ def update_user_reports(**kwargs):
...
+    # Logging values
+    total_reports = len(user_reports)
+    reports_with_event = 0
+    updated_reports = 0
+    samples = None
+
+    for project_id, reports in six.iteritems(project_map):
+        event_ids = [r.event_id for r in reports]
+        report_by_event = {r.event_id: r for r in reports}
+        snuba_filter = eventstore.Filter(project_ids=[project_id], event_ids=event_ids)
```
Comment (markstory) [inline src/sentry/tasks/update_user_reports.py]: Is there a time range we can apply to this query? Doing a query across the full retention window results in clickhouse not being able to use its indexes very well. Having a date condition will make these queries much faster.

**Review by markstory — APPROVED** [review]

File: src/sentry/tasks/update_user_reports.py (hunk 3 above)
Comment (evanh) [reply src/sentry/tasks/update_user_reports.py]: Yeah probably. I'll add one.

**Pull request merged**